From a3686315a604d06de6ed3976409bf39a3cb850fc Mon Sep 17 00:00:00 2001
From: Hyun Kwon <hyun.kwon@xilinx.com>
Date: Fri, 5 May 2017 17:29:55 -0700
Subject: [PATCH 032/532] drm: xilinx: dp: the M audio value should be in Khz

commit 68d5d95a5a3fe3f9d5e1a4f188001a03ff765664 from
https://github.com/Xilinx/linux-xlnx.git

In sync mode, the M audio value should be the audio clock rate
in Khz. Fix the calculation for that. While at it, remove
the test for debug print, and make it print the audio rate
in debug mode.

Signed-off-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/gpu/drm/xilinx/xilinx_drm_dp.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/xilinx/xilinx_drm_dp.c b/drivers/gpu/drm/xilinx/xilinx_drm_dp.c
index 7f201cb..3489fb9 100644
--- a/drivers/gpu/drm/xilinx/xilinx_drm_dp.c
+++ b/drivers/gpu/drm/xilinx/xilinx_drm_dp.c
@@ -1165,15 +1165,14 @@ static void xilinx_drm_dp_mode_set_stream(struct xilinx_drm_dp *dp,
 		xilinx_drm_writel(dp->iomem, XILINX_DP_TX_N_VID, reg);
 		xilinx_drm_writel(dp->iomem, XILINX_DP_TX_M_VID, mode->clock);
 		if (dp->aud_clk) {
-			int aud_rate = clk_get_rate(dp->aud_clk) / 512;
+			int aud_rate = clk_get_rate(dp->aud_clk);
 
-			if (aud_rate != 44100 && aud_rate != 48000)
-				dev_dbg(dp->dev, "Audio rate: %d\n", aud_rate);
+			dev_dbg(dp->dev, "Audio rate: %d\n", aud_rate / 512);
 
 			xilinx_drm_writel(dp->iomem, XILINX_DP_TX_AUDIO_N_AUD,
 					  reg);
 			xilinx_drm_writel(dp->iomem, XILINX_DP_TX_AUDIO_M_AUD,
-					  aud_rate);
+					  aud_rate / 1000);
 		}
 	}
 
-- 
2.7.4

